src/realm/column_binary.hpp

```diff
@@ -200,7 +200,7 @@ inline size_t BinaryColumn::size() const noexcept
         return leaf->size();
     }
     // Non-leaf root
-    return m_array->get_bptree_size();
+    return static_cast<BpTreeNode*>(m_array.get())->get_bptree_size();
```

Comment:
Oh now I remember the details: The point is also to debug-assert with a `dynamic_cast` before `static_cast`ing, to be absolutely sure that we don't `static_cast` something that isn't actually related. Just an idea. :)